components/EmotionCard.js

```diff
@@ -3,21 +3,28 @@ import BookmarkButton from "./BookmarkButton";
 import Link from "next/link";
 import EmotionCardContent from "./EmotionCardContent";
 import formatDate from "./TransformDateTime";
+import { emotionList } from "@/lib/emotionList";
 
 export default function EmotionCard({
   emotion,
   onToggleBookmark,
   isBookmarked,
   isDetailsPage = false,
-  intensity,
 }) {
   const { date, time } = formatDate(emotion.dateTime);
 
+  function getEmotionColor(type) {
+    const emotion = emotionList.find((item) => item.emotionType === type);
+    return emotion ? emotion.color : "var(--color-frame)";
+  }
```

Comment:
If you need this function also in another component, you could move it outside this component in a separate file (for example in a `utils` folder).